### ISSUE TITLE
Data leakage

### ISSUE THREAD

**Issue by dtischler95** [ChakradharG/PeptideBERT#10]
Hi,
first of all — thank you for making this project publicly available! 🙏 I noticed what seems to be a data leakage issue between your training, validation, and/or test datasets. This can lead to overly optimistic performance results and may compromise the reliability of the evaluation.
The dataset from WhiteLab contains duplicate sequence entries, and since the project uses a random train_test_split (with shuffling enabled), there is a significant risk that identical sequences may be distributed across different splits — such as training, validation, and test sets. 
To test this i used this Code Snippet after your `load_data()` function inside the file `train.py` after line 53:

```python


    tokenizer = BertTokenizer.from_pretrained('Rostlab/prot_bert_bfd', do_lower_case=False)
    train_sequences = [''.join(tokenizer.decode(row, skip_special_tokens=True).split(' ')) for row in
                       train_data_loader.dataset.input_ids] if train_data_loader is not None else []
    val_sequences = [''.join(tokenizer.decode(row, skip_special_tokens=True).split(' ')) for row in
                     val_data_loader.dataset.input_ids] if val_data_loader is not None else []
    test_sequences = [''.join(tokenizer.decode(row, skip_special_tokens=True).split(' ')) for row in
                      test_data_loader.dataset.input_ids] if test_data_loader is not None else []



    train_val_leakage = set(train_sequences) & set(val_sequences)
    train_test_leakage = set(train_sequences) & set(test_sequences)
    val_test_leakage = set(val_sequences) & set(test_sequences)

    print(f"Number of Training Data Points: {len(train_sequences)}")
    print(f"Number of Validation Data Points: {len(val_sequences)}")
    print(f"Number of Test Data Points: {len(test_sequences)}\n")

    if len(train_test_leakage) or len(train_val_leakage) or len(val_test_leakage) > 0:
        print("\n-------------- Data Leakage Detected --------------\n")

        if len(train_val_leakage) > 0:
            print(f"Number of data Points leaked in Training and Validation Data: {len(train_val_leakage)}\n")
        if len(train_test_leakage) > 0:
            print(f"Number of data Points leaked in Training and Test Data: {len(train_test_leakage)}\n")
        if len(val_test_leakage) > 0:
            print(f"Number of data Points leaked in Validation and Test Data: {len(val_test_leakage)}\n")

        print("\n-------------- Data Leakage Detected --------------\n")

    else:
        print("\n-------------- No Data Leakage Detected --------------\n")
```

You might want to consider deduplicating the dataset before splitting, or — if keeping duplicates is intentional — perform the train_test_split on the level of unique sequence IDs and sort the IDs afterwards to the intented Dataset This ensures that all occurrences of a given sequence remain confined to a single split (train/val/test), which prevents data leakage while preserving duplicates if needed.
